aws/src/main/java/org/apache/iceberg/aws/s3/S3InputStream.java

```diff
@@ -57,6 +64,14 @@ class S3InputStream extends SeekableInputStream implements RangeReadable {
   private final Counter readOperations;
 
   private int skipSize = 1024 * 1024;
+  private RetryPolicy<Object> retryPolicy =
+      RetryPolicy.builder()
+          .handle(
+              ImmutableList.of(
+                  SSLException.class, SocketTimeoutException.class, SocketException.class))
+          .onFailure(failure -> openStream(true))
+          .withMaxRetries(3)
```

Comment:
We can make these configurable down the line if needed but this covers all the cases that have been reported so far